packages/platform-browser/src/dom/shared_styles_host.ts

```diff
@@ -20,94 +20,159 @@ import {
 /** The style elements attribute name used to set value of `APP_ID` token. */
 const APP_ID_ATTRIBUTE_NAME = 'ng-app-id';
 
+/**
+ * A record of usage for a specific style including all elements added to the DOM
+ * that contain a given style.
+ */
+interface UsageRecord<T> {
+  elements: T[];
+  usage: number;
+}
+
+/**
+ * Removes all provided elements from the document.
+ * @param elements An array of HTML Elements.
+ */
+function removeAll(elements: Iterable<HTMLElement>): void {
+  for (const element of elements) {
+    element.remove();
+  }
+}
+
 @Injectable()
 export class SharedStylesHost implements OnDestroy {
-  // Maps all registered host nodes to a list of style nodes that have been added to the host node.
-  private readonly styleRef = new Map<
-    string /** Style string */,
-    {
-      elements: HTMLStyleElement[];
-      usage: number;
-    }
-  >();
-  private readonly hostNodes = new Set<Node>();
-  private readonly styleNodesInDOM: Map<string, HTMLStyleElement> | null;
-  private readonly platformIsServer: boolean;
+  /**
+   * Provides usage information for embedded style content and the HTML elements containing each style.
+   */
+  private readonly embeddedStyles = new Map<string /** content */, UsageRecord<HTMLStyleElement>>();
```

Comment:
nit: the "embeddedStyles" look good, but it's a bit harder to understand what they represent without looking at other code (how they are being used) or field description. I'm wondering if "currentStyles" might be a bit easier to understand:
```suggestion
  private readonly currentStyles = new Map<string /** content */, UsageRecord<HTMLStyleElement>>();
```

Reply:
I considered some different names for this as well. There are the SSR added styles which are also current.  I considered inline as well but the SSR added styles are also inline. So I ended up using the styles "origin" for the naming.  server for the SSR styles and embedded because they are inside the emitted JS for the component.